space_packet_parser/packets.py

```diff
@@ -157,6 +172,165 @@ def user_data(self) -> dict:
         return dict(list(self.items())[7:])
 
 
+class PacketCollection(list):
+    """Stores a list of packets."""
+    def __init__(
+        self,
+        packets: Iterable[CCSDSPacket],
+        *,
+        # TODO: Figure out typing with imports from definitions causing circular imports
+        # definitions.XtcePacketDefinition | None
+        packet_definition=None,
+    ):
+        """
+        Create a PacketCollection.
+
+        Parameters
+        ----------
+        apid_dict : dict
+            Mapping of APID to a list of packets with that apid.
+        packet_definition : XtcePacketDefinition
+            The packet definition to use for this collection.
+        """
+        super().__init__(packets)
+        self.packet_definition = packet_definition
+
+    def __str__(self):
+        apids = Counter(packet["PKT_APID"] for packet in self)
+        return (f"<PacketCollection>: {len(self)} packets\n"
+                + "Packets per apid (apid: npackets)\n"
+                + "\n".join(f"  {apid}: {count}" for apid, count in apids.items()))
+
+    @classmethod
+    def from_packet_file(
+        cls,
+        packet_file: str | Path,
+        # TODO: Figure out typing with imports from definitions causing circular imports
+        # str | Path | definitions.XtcePacketDefinition | None
+        packet_definition=None,
+    ) -> "PacketCollection":
+        """
+        Create a PacketCollection from a packet file.
+
+        Parameters
+        ----------
+        packet_file : str
+            Path to a file containing CCSDS packets.
+        packet_definition : str or Path or XtcePacketDefinition, optional
+            XTCE packet definition, or the path to the XTCE packet definition file.
+
+        Returns
+        -------
+        packet_collection : PacketCollection
+            A list of packets grouped together.
+        """
+        # TODO: Bring this import to the top of the file once circular dependencies are resolved
+        from space_packet_parser import definitions
+        if packet_definition is not None and not isinstance(packet_definition, definitions.XtcePacketDefinition):
+            # We got the path to a packet definition, so read it in
+            packet_definition = definitions.XtcePacketDefinition(packet_definition)
+
+        with open(packet_file, "rb") as binary_data:
+            # packet_generator = packets.packet_generator(binary_data, definition=packet_definition)
+            packet_generator = packet_definition.packet_generator(binary_data)
+            return cls(packet_generator, packet_definition=packet_definition)
+
+    def to_numpy(self, variable, raw_value=False):
```

Comment:
Do we want this to output a recarray/multi-dimensional array based on variables in the packets similar to the xarray dataset (i.e. shape: (npackets, nvariables))? Right now, I made it just do one specific variable a user requests.